pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -525,6 +542,21 @@ func (r *ReconcileBareMetalHost) actionInspecting(prov provisioner.Provisioner,
 		return recordActionFailure(info, metal3v1alpha1.InspectionError, provResult.ErrorMessage)
 	}
 
+	// Delete inspect annotation if exists
+	if hasInspectAnnotation(info.host) {
+		if info.host.Status.Provisioning.State == metal3v1alpha1.StateInspecting {
+			clearInspectAnnotations(info.host)
+			if err := r.client.Update(context.TODO(), info.host); err != nil {
+				return actionError{errors.Wrap(err, "failed to remove inspect annotations from host")}
+			}
+			return actionContinueNoWrite{}
+		} else {
+			info.host.Status.Provisioning.State = metal3v1alpha1.StateInspecting
```

Comment:
State changes should only happen in the state machine code.

Reply:
Fixed, thanks.